lib/engine/g_1860/share_pool.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require_relative '../share_pool'
+
+module Engine
+  module G1860
+    class SharePool < SharePool
+      def buy_shares(entity, shares, exchange: nil, exchange_price: nil, swap: nil)
+        bundle = shares.is_a?(ShareBundle) ? shares : ShareBundle.new(shares)
+        if bundle.presidents_share && bundle.owner == self
+          bundle = ShareBundle.new(bundle.shares, bundle.corporation.share_percent)
+        end
+
+        super(entity, bundle, exchange: exchange, exchange_price: exchange_price, swap: swap)
+      end
+
+      def transfer_shares(bundle, to_entity,
```

Comment:
how much of this has changed? what did you need to do

Reply:
The original transfer_shares doesn't deal with selling or buying president's shares to/from the pool.